src/ripple/app/tx/impl/SusPay.cpp

```diff
@@ -342,33 +350,61 @@ SusPayFinish::preflight (PreflightContext const& ctx)
     if (static_cast<bool>(cb) != static_cast<bool>(fb))
         return temMALFORMED;
 
+    // Verify the transaction signature. If it doesn't work
+    // then don't do any more work.
+    {
+        auto const ret = preflight2 (ctx);
+        if (!isTesSuccess (ret))
+            return ret;
+    }
+
     if (cb && fb)
     {
         auto& router = ctx.app.getHashRouter();
 
         auto const id = ctx.tx.getTransactionID();
         auto const flags = router.getFlags (id);
 
-        // If we already checked the fulfillment and found
-        // it to be bad, don't bother checking again:
-        if (flags & SF_CONDBAD)
-            return temMALFORMED;
-
-        if (! (flags & SF_CONDGOOD))
+        // If we haven't checked the condition, check it
+        // now. Whether it passes or not isn't important
+        // in preflight.
+        if (!(flags & (SF_CF_INVALID | SF_CF_VALID)))
         {
-            if (!checkCondition (*fb, *cb))
-            {
-                router.setFlags (id, SF_CONDBAD);
-                return temMALFORMED;
-            }
+            if (checkCondition (*fb, *cb))
+                router.setFlags (id, SF_CF_VALID);
+            else
+                router.setFlags (id, SF_CF_INVALID);
+        }
+    }
+
+    return tesSUCCESS;
+}
 
-            router.setFlags (id, SF_CONDGOOD);
+std::uint64_t
+SusPayFinish::calculateBaseFee (
+    PreclaimContext const& ctx)
+{
+    std::uint64_t baseFee = Transactor::calculateBaseFee (ctx);
+    std::uint64_t const baseFee2 = baseFee;
+    std::uint64_t len = 0;
+
+    if (auto const cb = ctx.tx[~sfCondition])
+    {
+        baseFee += ctx.view.fees().units;
+
+        if (auto c = ripple::cryptoconditions::loadCondition(*cb))
```

Comment:
If `loadCondition` fails, should that imply a maximal fee be applied?  Or is that filtered earlier?